### PR TITLE
fix(upgrade-test): enable back the usage of sstabledump

### PR DESCRIPTION
**Pull request by fruch**
since in older versions i.e. 2023.1 and lower, we
can use `scylla sstable` tools yet, we should revert to use the old `sstabledump` tool

#### Testing
- [x] https://jenkins.scylladb.com/job/enterprise-2023.1/job/branch_2023.1_future/job/rolling-upgrade-ami-arm-test/3/

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by soyacz — APPROVED** [review]
if works, LGTM

**Review by juliayakovlev — APPROVED** [review]

**Comment by fruch** [comment]
> if works, LGTM

trying it out, we'll knew in few hours

File: upgrade_test.py [hunk 1]
```diff
@@ -765,11 +768,15 @@ def test_rolling_upgrade(self):  # pylint: disable=too-many-locals,too-many-stat
 
         # Verify sstabledump
         InfoEvent(message='Starting sstabledump to verify correctness of sstables').publish()
+        if ComparableScyllaVersion(self.db_cluster.nodes[0].scylla_version) < "2023.2" or \
+                ComparableScyllaVersion(self.db_cluster.nodes[0].scylla_version) < "5.4":
```
Comment (vponomaryov) [inline upgrade_test.py:772]: This condition will always provide `True` for any OSS version.
It is bug.
Need to reorder condition parts or add a check for enterprise-type in addition to existing conditions.

Comment (fruch) [reply upgrade_test.py:772]: good catch, as some point we might want to add something to ComparableScyllaVersion, to help doing those in less verbose manner

**Pull request merged**